lib/apiservers/engine/backends/eventmonitor.go

```diff
@@ -120,8 +120,21 @@ func (m *PortlayerEventMonitor) Start() error {
 	}
 
 	m.stop = make(chan struct{})
-	go m.monitor()
-
+	go func() {
+		var err error
+		for {
+			select {
+			case <-m.stop:
+				break
+			default:
+				if err = m.monitor(); err != nil {
+					log.Error(err)
+				}
+				log.Debug("Restarting Portlayer event monitor")
```

Comment:
I'd actually make this a `Warn` - it's not something that should happen under normal circumstances and is an indication that we may have missed significant state transitions.